importing bin/kc.[sh|bat] import --file <file> doesn't work when using external database

### Describe the bug

I am using external database MariaDB, and when I use importing using 'bin/kc.[sh|bat] import --file <file>' it doesnt work.

### Version

18.0.0

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

Can confirm.

I run `bin/kc.sh import --file data/import/SOME_FILE.json --override=true` in a Dockerfile right before  before starting Keycloak in the entrypoint and I get the following exception:
```
 WARN  [io.agroal.pool] (agroal-11) Datasource '<default>': Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.                                             
#0 1.956 2022-07-18 10:54:57,539 WARN  [org.hibernate.engine.jdbc.env.internal.JdbcEnvironmentInitiator] (JPA Startup Thread: keycloak-default) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: Connection to localhost:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
#0 1.956        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:319)
#0 1.956        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
#0 1.956        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:223)
#0 1.956        at org.postgresql.Driver.makeConnection(Driver.java:400)
#0 1.956        at org.postgresql.Driver.connect(Driver.java:259)
#0 1.956        at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:677)
#0 1.956        at java.sql/java.sql.DriverManager.getConnection(DriverManager.java:228)
#0 1.956        at org.postgresql.ds.common.BaseDataSource.getConnection(BaseDataSource.java:103)
#0 1.956        at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:49)
#0 1.956        at org.postgresql.xa.PGXADataSource.getXAConnection(PGXADataSource.java:35)
#0 1.956        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:216)
#0 1.956        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:513)
#0 1.956        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:494)
#0 1.956        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
#0 1.956        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
#0 1.956        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
#0 1.956        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
#0 1.956        at java.base/java.lang.Thread.run(Thread.java:829)
#0 1.956 Caused by: java.net.ConnectException: Connection refused (Connection refused)
#0 1.956        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
#0 1.956        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
#0 1.956        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:255)
#0 1.956        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
#0 1.956        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
#0 1.956        at java.base/java.net.Socket.connect(Socket.java:609)
#0 1.956        at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
#0 1.956        at org.postgresql.core.PGStream.<init>(PGStream.java:98)
#0 1.956        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
#0 1.956        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
#0 1.956        ... 17 more

```
Postgres was configured through `bin/kc.sh build --db postgres`. It appears that the import command ignored the environment variables and connects to the default  `localhost:5432`. However, `KC_DB_URL_HOST` is set.

Another interesting insight is that `bin/kc.[sh|bat] start --import-realm` works without problems and uses `KC_DB_URL_HOST` without problems.

--- a/docs/documentation/release_notes/topics/22_0_0.adoc
+++ b/docs/documentation/release_notes/topics/22_0_0.adoc
@@ -1,3 +1,10 @@
 = Legacy Promise API removed from Keycloak JS adapter
 
 With this release, we have removed the legacy Promise API methods from the Keycloak JS adapter. This means that calling `.success()` and `.error()` on promises returned from the adapter is no longer possible.
+
+= Export and Import perform an automatic build
+
+In previous releases, the `export` and `import` commands required a `build` command to be run first.
+Starting with this release, the `export` and `import` commands perform an automatic rebuild of Keycloak if a build time configuration has changed.
+
+See the migration guide for details.

--- a/docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
+++ b/docs/documentation/upgrading/topics/keycloak/changes-22_0_0.adoc
@@ -38,3 +38,39 @@ try {
   alert('failed to initialize');
 }
 ```
+
+= Export and Import perform an automatic build
+
+In previous releases, the `export` and `import` commands required a `build` command to be run first.
+Starting with this release, the `export` and `import` commands perform an automatic rebuild of Keycloak if a build time configuration has changed.
+
+When migrating existing scripts that run a `build` command first, migrate by adding the `--optimized` command line option to the `export` and `import` command to avoid Keycloak automatically re-building the image.
+Not adding the `--optimized` option in this might make Keycloak trigger a rebuild and revert to the default values, and then connecting to the database for export and import will not work.
+
+The following examples assume that runtime parameters like a database password are provided via a configuration file or an environment variable.
+
+.Before migration: Running the build command before running the export command
+[source,bash]
+----
+bin/kc.[sh|bat] build --db=postgres ...
+bin/kc.[sh|bat] export --dir <dir>
+----
+
+.After migration: Adding `--optimized` to the export command
+[source,bash,subs="+quotes"]
+----
+bin/kc.[sh|bat] build --db=postgres ...
+bin/kc.[sh|bat] export ##--optimized## --dir <dir>
+----
+
+.After migration: Leveraging the auto-build functionality
+[source,bash]
+----
+bin/kc.[sh|bat] export --dir <dir> --db=postgres ...
+----
+
+NOTE:: When the auto-build runs, the build time options will be in effect for all subsequent commands that are started with the `--optimized` flag, including the `start` command.
+
+In previous releases the `export` and `import` commands allowed runtime parameters like, for example, a database URL only in configuration files or environment variables.
+Starting with this release, those runtime parameters are now available on the command line as well.
+Use the `--help` option to find out about the supported parameters.

--- a/docs/guides/server/importExport.adoc
+++ b/docs/guides/server/importExport.adoc
@@ -1,11 +1,24 @@
 <#import "/templates/guide.adoc" as tmpl>
 <#import "/templates/kc.adoc" as kc>
+<#import "/templates/links.adoc" as links>
 
 <@tmpl.guide
     title="Importing and Exporting Realms"
     summary="An overview about how to import and export realms">
 
 In this guide, you are going to understand the different approaches for importing and exporting realms using JSON files.
+
+== Providing options for database connection parameters
+
+When using the `export` and the `import` commands below, Keycloak needs to know how to connect to the database where the information about realms, clients, users and other entities is stored.
+As described in <@links.server id="configuration"/> that information can be provided as command line parameters, environment variables or a configuration file.
+Use the `--help` command line option for each command to see the available options.
+
+Some of the configuration options are build time configuration options.
+As default, Keycloak will re-build automatically for the `export` and `import` commands if it detects a change of a build time parameter.
+
+If you have built an optimized version of Keycloak with the `build` command as outlined in <@links.server id="configuration"/>, use the command line option `--optimized` to have Keycloak skip the build check for a faster startup time.
+When doing this, remove the build time options from the command line and keep only the runtime options.
 
 == Exporting a Realm to a Directory
 

--- a/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJob.java
+++ b/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJob.java
@@ -165,7 +165,7 @@ public class KeycloakRealmImportJob extends OperatorManagedResource {
         var runBuild = (keycloak.getSpec().getImage() == null) ? "/opt/keycloak/bin/kc.sh build && " : "";
 
         var commandArgs = List.of("-c",
-                runBuild + "/opt/keycloak/bin/kc.sh import --file='" + importMntPath + getRealmName() + "-realm.json' " + override);
+                runBuild + "/opt/keycloak/bin/kc.sh import --optimized --file='" + importMntPath + getRealmName() + "-realm.json' " + override);
 
         keycloakContainer
                 .setCommand(command);

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/Picocli.java
@@ -21,6 +21,7 @@ import static org.keycloak.quarkus.runtime.Environment.isRebuildCheck;
 import static org.keycloak.quarkus.runtime.Environment.isRebuilt;
 import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.*;
 import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.AUTO_BUILD_OPTION_LONG;
+import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG;
 import static org.keycloak.quarkus.runtime.configuration.ConfigArgsConfigSource.parseConfigArgs;
 import static org.keycloak.quarkus.runtime.configuration.Configuration.OPTION_PART_SEPARATOR;
 import static org.keycloak.quarkus.runtime.configuration.Configuration.getBuildTimeProperty;
@@ -53,11 +54,9 @@ import org.keycloak.config.OptionCategory;
 import org.keycloak.quarkus.runtime.cli.command.AbstractCommand;
 import org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand;
 import org.keycloak.quarkus.runtime.cli.command.Build;
-import org.keycloak.quarkus.runtime.cli.command.Export;
-import org.keycloak.quarkus.runtime.cli.command.Import;
 import org.keycloak.quarkus.runtime.cli.command.ImportRealmMixin;
 import org.keycloak.quarkus.runtime.cli.command.Main;
-import org.keycloak.quarkus.runtime.cli.command.Start;
+import org.keycloak.quarkus.runtime.cli.command.ShowConfig;
 import org.keycloak.quarkus.runtime.cli.command.StartDev;
 import org.keycloak.quarkus.runtime.configuration.ConfigArgsConfigSource;
 import org.keycloak.quarkus.runtime.configuration.PersistedConfigSource;
@@ -117,7 +116,7 @@ public final class Picocli {
                 Environment.forceDevProfile();
             }
         }
-        if (requiresReAugmentation(cmd)) {
+        if (requiresReAugmentation(getCurrentCommandSpec(cliArgs, cmd.getCommandSpec()))) {
             exitCode = runReAugmentation(cliArgs, cmd);
         }
 
@@ -128,12 +127,11 @@ public final class Picocli {
         return cliArgs.contains("--help")
                 || cliArgs.contains("-h")
                 || cliArgs.contains("--help-all")
-                || cliArgs.contains(Export.NAME)
-                || cliArgs.contains(Import.NAME);
+                || cliArgs.contains(ShowConfig.NAME);
     }
 
-    public static boolean requiresReAugmentation(CommandLine cmd) {
-        if (hasConfigChanges()) {
+    public static boolean requiresReAugmentation(CommandLine cmdCommand) {
+        if (hasConfigChanges(cmdCommand)) {
             if (!ConfigArgsConfigSource.getAllCliArgs().contains(StartDev.NAME) && "dev".equals(getConfig().getOptionalValue("kc.profile", String.class).orElse(null))) {
                 return false;
             }
@@ -174,17 +172,17 @@ public final class Picocli {
             cmd.getOut().println("Changes detected in configuration. Updating the server image.");
         }
 
-        int exitCode = 0;
+        int exitCode;
 
         List<String> configArgsList = new ArrayList<>(cliArgs);
 
-        configArgsList.replaceAll(Picocli::replaceStartWithBuild);
+        configArgsList.replaceAll(arg -> replaceCommandWithBuild(getCurrentCommandSpec(cliArgs, cmd.getCommandSpec()).getCommandName(), arg));
         configArgsList.removeIf(Picocli::isRuntimeOption);
 
         exitCode = cmd.execute(configArgsList.toArray(new String[0]));
 
         if(!isDevMode() && exitCode == cmd.getCommandSpec().exitCodeOnSuccess()) {
-            cmd.getOut().printf("Next time you run the server, just run:%n%n\t%s %s %s %s%n%n", Environment.getCommand(), Start.NAME, OPTIMIZED_BUILD_OPTION_LONG, String.join(" ", getSanitizedRuntimeCliOptions()));
+            cmd.getOut().printf("Next time you run the server, just run:%n%n\t%s %s %s %s%n%n", Environment.getCommand(), getCurrentCommandSpec(cliArgs, cmd.getCommandSpec()).getCommandName(), OPTIMIZED_BUILD_OPTION_LONG, String.join(" ", getSanitizedRuntimeCliOptions()));
         }
 
         return exitCode;
@@ -222,7 +220,7 @@ public final class Picocli {
         return false;
     }
 
-    private static boolean hasConfigChanges() {
+    private static boolean hasConfigChanges(CommandLine cmdCommand) {
         Optional<String> currentProfile = Optional.ofNullable(Environment.getProfile());
         Optional<String> persistedProfile = getBuildTimeProperty("kc.profile");
 
@@ -249,6 +247,17 @@ public final class Picocli {
 
             String persistedValue = getBuildTimeProperty(propertyName).orElse("");
             String runtimeValue = getRuntimeProperty(propertyName).orElse(null);
+
+            // compare only the relevant options for this command, as not all options might be set for this command
+            if (cmdCommand.getCommand() instanceof AbstractCommand) {
+                AbstractCommand abstractCommand = cmdCommand.getCommand();
+                PropertyMapper mapper = PropertyMappers.getMapper(propertyName);
+                if (mapper != null) {
+                    if (!abstractCommand.getOptionCategories().contains(mapper.getCategory())) {
+                        continue;
+                    }
+                }
+            }
 
             if (runtimeValue == null && isNotBlank(persistedValue)) {
                 PropertyMapper mapper = PropertyMappers.getMapper(propertyName);
@@ -375,7 +384,7 @@ public final class Picocli {
 
             if (!includeBuildTime && !includeRuntime) {
                 return;
-            } else if (includeRuntime && !includeBuildTime && (Start.NAME.equals(command.getCommandName())) || StartDev.NAME.equals(command.getCommandName())) {
+            } else if (includeRuntime && !includeBuildTime && !ShowConfig.NAME.equals(command.getCommandName())) {
                 includeBuildTime = isRebuilt() || !cliArgs.contains(OPTIMIZED_BUILD_OPTION_LONG);
             } else if (includeBuildTime && !includeRuntime) {
                 includeRuntime = isRebuildCheck();
@@ -519,8 +528,8 @@ public final class Picocli {
         return args;
     }
 
-    private static String replaceStartWithBuild(String arg) {
-        if (arg.equals(Start.NAME) || arg.equals(StartDev.NAME)) {
+    private static String replaceCommandWithBuild(String commandName, String arg) {
+        if (arg.equals(commandName)) {
             return Build.NAME;
         }
         return arg;

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractExportImportCommand.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/AbstractExportImportCommand.java
@@ -29,6 +29,9 @@ public abstract class AbstractExportImportCommand extends AbstractStartCommand i
     private final String action;
 
     @CommandLine.Mixin
+    OptimizedMixin optimizedMixin;
+
+    @CommandLine.Mixin
     HelpAllMixin helpAllMixin;
 
     protected AbstractExportImportCommand(String action) {
@@ -51,6 +54,9 @@ public abstract class AbstractExportImportCommand extends AbstractStartCommand i
                         optionCategory != OptionCategory.PROXY &&
                         optionCategory != OptionCategory.HOSTNAME &&
                         optionCategory != OptionCategory.METRICS &&
+                        optionCategory != OptionCategory.VAULT &&
+                        optionCategory != OptionCategory.SECURITY &&
+                        optionCategory != OptionCategory.CACHE &&
                         optionCategory != OptionCategory.HEALTH).collect(Collectors.toList());
     }
 

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Build.java
@@ -34,7 +34,6 @@ import picocli.CommandLine;
 import picocli.CommandLine.Command;
 
 import java.util.List;
-import java.util.stream.Collectors;
 
 @Command(name = Build.NAME,
         header = "Creates a new and optimized server image.",
@@ -91,7 +90,8 @@ public final class Build extends AbstractCommand implements Runnable {
     }
 
     public List<OptionCategory> getOptionCategories() {
-        return super.getOptionCategories().stream().filter(optionCategory -> optionCategory != OptionCategory.EXPORT && optionCategory != OptionCategory.IMPORT).collect(Collectors.toList());
+        // all options should work for the build command, otherwise re-augmentation might fail due to unknown options
+        return super.getOptionCategories();
     }
 
     private void exitWithErrorIfDevProfileIsSetAndNotStartDev() {

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/OptimizedMixin.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/OptimizedMixin.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.quarkus.runtime.cli.command;
+
+import picocli.CommandLine;
+
+import static org.keycloak.quarkus.runtime.cli.Picocli.NO_PARAM_LABEL;
+import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG;
+
+public final class OptimizedMixin {
+
+    @CommandLine.Option(names = {OPTIMIZED_BUILD_OPTION_LONG},
+            description = "Use this option to achieve an optimal startup time if you have previously built a server image using the 'build' command.",
+            paramLabel = NO_PARAM_LABEL,
+            order = 1)
+    Boolean optimized;
+
+}

--- a/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java
+++ b/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java
@@ -19,6 +19,7 @@ package org.keycloak.quarkus.runtime.cli.command;
 
 import static org.keycloak.quarkus.runtime.Environment.setProfile;
 import static org.keycloak.quarkus.runtime.cli.Picocli.NO_PARAM_LABEL;
+import static org.keycloak.quarkus.runtime.cli.command.AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG;
 import static org.keycloak.quarkus.runtime.configuration.Configuration.getRawPersistedProperty;
 
 import org.keycloak.config.OptionCategory;
@@ -37,8 +38,8 @@ import java.util.stream.Collectors;
         description = {
                 "%nUse this command to run the server in production."
         },
-        footer = "%nBy default, this command tries to update the server configuration by running a '" + Build.NAME + "' before starting the server. You can disable this behavior by using the '" + Start.OPTIMIZED_BUILD_OPTION_LONG + "' option:%n%n"
-                + "      $ ${PARENT-COMMAND-FULL-NAME:-$PARENTCOMMAND} ${COMMAND-NAME} '" + Start.OPTIMIZED_BUILD_OPTION_LONG + "'%n%n"
+        footer = "%nBy default, this command tries to update the server configuration by running a '" + Build.NAME + "' before starting the server. You can disable this behavior by using the '" + OPTIMIZED_BUILD_OPTION_LONG + "' option:%n%n"
+                + "      $ ${PARENT-COMMAND-FULL-NAME:-$PARENTCOMMAND} ${COMMAND-NAME} '" + OPTIMIZED_BUILD_OPTION_LONG + "'%n%n"
                 + "By doing that, the server should start faster based on any previous configuration you have set when manually running the '" + Build.NAME + "' command.")
 public final class Start extends AbstractStartCommand implements Runnable {
 
@@ -52,11 +53,8 @@ public final class Start extends AbstractStartCommand implements Runnable {
             order = 1)
     Boolean autoConfig;
 
-    @CommandLine.Option(names = {OPTIMIZED_BUILD_OPTION_LONG},
-            description = "Use this option to achieve an optional startup time if you have previously built a server image using the 'build' command.",
-            paramLabel = NO_PARAM_LABEL,
-            order = 1)
-    Boolean optimized;
+    @CommandLine.Mixin
+    OptimizedMixin optimizedMixin;
 
     @CommandLine.Mixin
     ImportRealmMixin importRealmMixin;

--- a/quarkus/runtime/src/main/resources/META-INF/keycloak.conf
+++ b/quarkus/runtime/src/main/resources/META-INF/keycloak.conf
@@ -22,6 +22,7 @@ metrics-enabled=false
 %import_export.http-server-enabled=false
 %import_export.hostname-strict=false
 %import_export.hostname-strict-https=false
+%import_export.cache=local
 
 #logging defaults
 log-console-output=default

--- a/quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/BasicDatabaseTest.java
+++ b/quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/BasicDatabaseTest.java
@@ -38,7 +38,7 @@ public abstract class BasicDatabaseTest {
     }
 
     @Test
-    @Launch({ "start", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG,"--http-enabled=true", "--hostname-strict=false", "--db-username=wrong" })
+    @Launch({ "start", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG, "--http-enabled=true", "--hostname-strict=false", "--db-username=wrong" })
     void testWrongUsername(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertMessage("ERROR: Failed to obtain JDBC connection");
@@ -48,7 +48,7 @@ public abstract class BasicDatabaseTest {
     protected abstract void assertWrongUsername(CLIResult cliResult);
 
     @Test
-    @Launch({ "start", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG,"--http-enabled=true", "--hostname-strict=false", "--db-password=wrong" })
+    @Launch({ "start", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG, "--http-enabled=true", "--hostname-strict=false", "--db-password=wrong" })
     void testWrongPassword(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertMessage("ERROR: Failed to obtain JDBC connection");
@@ -59,7 +59,7 @@ public abstract class BasicDatabaseTest {
 
     @Order(1)
     @Test
-    @Launch({ "export", "--dir=./target/export"})
+    @Launch({ "export", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG, "--dir=./target/export"})
     public void testExportSucceeds(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertMessage("Full model export requested");
@@ -68,7 +68,7 @@ public abstract class BasicDatabaseTest {
 
     @Order(2)
     @Test
-    @Launch({ "import", "--dir=./target/export" })
+    @Launch({ "import", AbstractStartCommand.OPTIMIZED_BUILD_OPTION_LONG, "--dir=./target/export" })
     void testImportSucceeds(LaunchResult result) {
         CLIResult cliResult = (CLIResult) result;
         cliResult.assertMessage("target/export");

